x-pack/plugins/canvas/public/functions/timelion.js

```diff
@@ -88,14 +71,10 @@ export const timelion = () => ({
       },
     };
 
-    return fetch(`${handlers.serverUri}/api/timelion/run`, {
+    return fetch(chrome.addBasePath(`/api/timelion/run`), {
```

Comment:
I'd recommend replacing `fetch` with `kfetch` by:

```
import { kfetch } from 'ui/kfetch'

```

And then applying the following patch:
```
diff --git a/x-pack/plugins/canvas/public/functions/timelion.js b/x-pack/plugins/canvas/public/functions/timelion.js
index 3f803e4679..1879af4849 100644
--- a/x-pack/plugins/canvas/public/functions/timelion.js
+++ b/x-pack/plugins/canvas/public/functions/timelion.js
@@ -5,8 +5,8 @@
  */
 
 import chrome from 'ui/chrome';
+import { kfetch } from 'ui/kfetch';
 import { flatten } from 'lodash';
-import { fetch } from '../../common/lib/fetch';
 import { buildBoolArray } from '../../server/lib/build_bool_array';
 
 export const timelion = () => ({
@@ -71,12 +71,13 @@ export const timelion = () => ({
       },
     };
 
-    return fetch(chrome.addBasePath(`/api/timelion/run`), {
+    return kfetch({
+      pathname: `/api/timelion/run`,
       method: 'POST',
       responseType: 'json',
-      data: body,
+      body: JSON.stringify(body),
     }).then(resp => {
-      const seriesList = resp.data.sheet[0].list;
+      const seriesList = resp.sheet[0].list;
 
       const rows = flatten(
         seriesList.map(series =>
```

Reply:
I was hoping to defer that and update other uses all in one shot, but OK.